marshmallow_recipe/serialization.py

```diff
@@ -151,3 +151,15 @@ def dump_many_v2(data: list[Any], *, naming_case: NamingCase | None = None) -> l
     dump_many = dump_many_v2
 
 EmptySchema = m.Schema
+
+
+def _extract_type(data: Any, cls: type | None) -> type:
+    if hasattr(data, "__orig_class__"):
+        return getattr(data, "__orig_class__")
```

Comment:
as far as `__orig_class__` is an implementation detail of CPython, it makes sense to use following checks of this method for it's value as well - basically, to rewrite it in the following way:
```python
   if hasattr(data, "__orig_class__"):
        cls = getattr(data, "__orig_class__")
```

Reply:
please check the latest implementation https://github.com/anna-money/marshmallow-recipe/pull/163/files#diff-f612e661488039f541bcbadfa92967e0b6c7f6207fb0fb85d144a65ee3f03aacR21